src/httpAgent.js

```diff
@@ -0,0 +1,58 @@
+import * as logger from './logger';
+import { TracerGlobals } from './globals';
+import { getEdgeUrl, getJSONBase64Size, getTracerInfo } from './utils';
+import axios from 'axios';
+import http from 'http';
+import https from 'https';
+
+export const HttpAgent = (() => {
```

Comment:
As I understand, this file isn't relevant to HTTP hooks, it's only for reporting the spans. Maybe rename it so it will be easier to understand.